### PR TITLE
Benji/jib 78 add invitation loading screen

### PR DESCRIPTION
**Pull request by Benjamindodgson**

### CODE REVIEW

File: Shared/Models/Parse Subclasses/Connection.swift [hunk 1]
```diff
@@ -47,6 +48,11 @@ final class Connection: PFObject, PFSubclassing {
         return self.getObject(for: .conversationSid)
     }
 
+    var initialConversations: [String] {
```
Comment (Benjamindodgson) [inline Shared/Models/Parse Subclasses/Connection.swift:51]: Will be used by the backend to add all new users to every conversation in this array.

File: iOS/Flows/Conversation/ConversationCoordinator.swift [hunk 2]
```diff
@@ -71,12 +71,53 @@ class ConversationCoordinator: PresentableCoordinator<Void> {
...
-        self.addChildAndStart(coordinator) { result in
+
+        self.addChildAndStart(coordinator) { connections in
             coordinator.toPresentable().dismiss(animated: true)
+            self.add(connections: connections, to: self.conversationVC.conversationController)
         }
         self.router.present(coordinator, source: self.conversationVC)
     }
 
+    func add(connections: [Connection], to controller: ChatChannelController) {
```
Comment (Benjamindodgson) [inline iOS/Flows/Conversation/ConversationCoordinator.swift:82]: Where we handle all connections returned from the people controller.

File: iOS/Flows/People/PeopleCoordinator.swift [hunk 3]
```diff
@@ -10,9 +10,28 @@ import Foundation
 import StreamChat
 import Contacts
 
-class PeopleCoordinator: PresentableCoordinator<ChatChannelController?> {
+class PeopleCoordinator: PresentableCoordinator<[Connection]> {
```
Comment (Benjamindodgson) [inline iOS/Flows/People/PeopleCoordinator.swift:13]: Now returns any newly created, or existing connections.

File: iOS/Flows/Reservations/ReservationsCoordinator.swift [hunk 4]
```diff
@@ -15,14 +15,6 @@ import Combine
 class ReservationsCoordinator: PresentableCoordinator<Void> {
 
     lazy var reservationsVC = ReservationsViewController()
-    private var selectedContact: CNContact?
```
Comment (Benjamindodgson) [inline iOS/Flows/Reservations/ReservationsCoordinator.swift:18]: Moved all this to the peoplecoordinator

File: iOS/Models/Contact.swift [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+//
+//  Created by Benji Dodgson on 9/28/21.
+//  Copyright © 2021 Benjamin Dodgson. All rights reserved.
+//
+
+import Foundation
+import Contacts
+import PhoneNumberKit
+
+struct Contact: Avatar, Hashable {
```
Comment (Benjamindodgson) [inline iOS/Models/Contact.swift:13]: Used to show what contacts have already been invited. UI not implemented yet to show this.

**Review by MartinJibber — APPROVED** [review]
Sweet

**Pull request merged**